src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,96 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import java.util.Arrays;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User("validLogin", "password123", 25);
+        Storage.people.addAll(Arrays.asList(new User("Bob123", "password123", 20),
+                new User("Alice231", "password123", 25),
+                new User("Vladimir295", "password123", 25),
+                new User("Semen294", "short", 25),
+                new User("Pavel194", "password123", 17),
+                new User("Joe0192", "validPass456", 30)));
```

Comment:
Directly adding users to `Storage.people` in the `setUp` method with `Storage.people.addAll()` is not recommended because it relies on the state of the storage which can be modified by other tests. Instead, you should add necessary users individually in each test where they are needed to ensure that tests are independent.